### PR TITLE
add image-level feature support

### PR DESCRIPTION
**Pull request by Lamply**
Resize Cx1x1 feature map to CxHxW, use in DeepLabv3 etc.

### CODE REVIEW

File: src/layer/interp.cpp [hunk 1]
```diff
@@ -56,26 +62,42 @@ int Interp::forward(const Mat &bottom_blob, Mat &top_blob) const
     if (top_blob.empty())
         return -100;
 
-    if (resize_type == 1)//nearest
+    if (resize_type == 1 || bottom_blob.dims == 1)//nearest
```
Comment (nihui) [inline src/layer/interp.cpp]: If bottom_blob is a vector, the resize_type is ignored. move this condition out of resize_type path.

```
if (bottom_blob.dims == 1)
{
    // handle this special case, ignore resize_type

    return 0;
}

// all the following code shall not be changed
if (resize_type == 1)//nearest
{
}

```

File: src/layer/interp.cpp [hunk 2]
```diff
@@ -56,26 +62,42 @@ int Interp::forward(const Mat &bottom_blob, Mat &top_blob) const
...
-        #pragma omp parallel for
-        for (int q = 0; q < c; ++q)
+        if(bottom_blob.dims == 1)
+        {
+            #pragma omp parallel for
+            for (int q = 0; q < c; ++q)
+            {
+                const float *ptr = ((float*)bottom_blob.data+q);
+                float *output_ptr = top_blob.channel(q);
+                for (int i = 0; i < top_blob.cstep; ++i)
```
Comment (nihui) [inline src/layer/interp.cpp]: This is what the Mat::fill() does

File: src/layer/interp.cpp [hunk 3]
```diff
@@ -56,26 +62,38 @@ int Interp::forward(const Mat &bottom_blob, Mat &top_blob) const
     if (top_blob.empty())
         return -100;
 
+	if (bottom_blob.dims == 1)
+	{
+		#pragma omp parallel for
```
Comment (nihui) [inline src/layer/interp.cpp]: always use space, not tab

File: src/layer/interp.cpp [hunk 4]
```diff
@@ -56,26 +62,38 @@ int Interp::forward(const Mat &bottom_blob, Mat &top_blob) const
...
+		return 0;
+	}
+	
     if (resize_type == 1)//nearest
     {
-        #pragma omp parallel for
-        for (int q = 0; q < c; ++q)
-        {
-            const float *ptr = bottom_blob.channel(q);
-            float *output_ptr = top_blob.channel(q);
```
Comment (nihui) [inline src/layer/interp.cpp]: keep the ident space of this block

File: src/layer/interp.cpp (hunk 3 above)
Comment (Lamply) [reply src/layer/interp.cpp]: Oh, thanks

**Pull request merged**